### PR TITLE
Nukes role restrictions

### PR DESCRIPTION
**Pull request by VigersRay**
## About the PR

If you don't know how to turn on the harm mod, you don't play nukers.
The leader must play 5 hours in command and sec department, he must be able to lead and fight. 
Medic must play 5 hours in the medical department, he must be able to heal.
A standard nuker should also play 5 hours in sec department. 
And to know the general aspects of the game, all hes need to play 25 hours.

Depends:
https://github.com/space-wizards/space-station-14/pull/17928

**Media**

- [ ] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

:cl: VigersRay
- tweak: The syndicate began to recruit more experienced people into the ranks of nuclear operatives.

### CODE REVIEW

**Review by crazybrain23 — CHANGES_REQUESTED** [review]

**Comment by Geekyhobo** [comment]
Personally I think this is going too far, the whole reason that #17888 was merged was because it would allow more people to paly nukie. I get that its frustrating, especially as a nukie, when your commander or teammates seem like they have never seen a gun before in their life. But its not fair to people just to gatekeep a fun and enjoyable role. Plus this technically makes it rarer for people to get the roles. Example being, if there is only 2 people that have the requirements to be the commander on a server they will get the role, unless your are able to also ignore those role time requirements. Also in the same line of thinking, if there is no one with the capability of filling the nukie roles, does that mean that a nukie round can never start? 

Idk that was a bit of a rant but I mean it as constructive criticism. I think this could be a good idea, just need to be fleshed out a bit more than just putting role timers on it.

**Comment by HoidC** [comment]
> Personally I think this is going too far, the whole reason that #17888 was merged was because it would allow more people to paly nukie. I get that its frustrating, especially as a nukie, when your commander or teammates seem like they have never seen a gun before in their life. But its not fair to people just to gatekeep a fun and enjoyable role. Plus this technically makes it rarer for people to get the roles. Example being, if there is only 2 people that have the requirements to be the commander on a server they will get the role, unless your are able to also ignore those role time requirements. Also in the same line of thinking, if there is no one with the capability of filling the nukie roles, does that mean that a nukie round can never start?
> 
> Idk that was a bit of a rant but I mean it as constructive criticism. I think this could be a good idea, just need to be fleshed out a bit more than just putting role timers on it.

the command timers are a bit much but this role absolutely needs a playtime limiter for how difficult it is 

there's been multiple instances that i've personally witnessed of players having their first round be a nuke op round which is going to suck for them and everyone else around them - they're inevitably going to be yelled at by their own team and laughed at in OOC chat when they die, and they're also thrown into the most complex role in the game

roles that are difficult have role timers so that players can learn them before they play them and this shouldn't be different

**Comment by deltanedas** [comment]
i think:
- commander should require a few hours as operative
- operative needs salvage time
- theyre still low times

**Comment by LankLTE** [comment]
> i think:
> 
>     * commander should require a few hours as operative
> 
>     * operative needs salvage time
> 
>     * theyre still low times

I agree commander should be operative time, but afaik that's not tracked so it'd need to be worked on.
Why should operative need salvage time..? They're not even remotely related roles. Simple playtime requirement should be enough for salvage.
The times are somewhat short (and nukie needing 5 hours of command for whatever reason is odd)- personally like 50 hours to get nukie should be the max though. It's a very unique role and the truth is you'll never be able to get rid of new players being nukie entire, just lower it.

**Comment by GoodWheatley** [comment]
Commander requiring operative time would suck because of how low the chances of rolling nukie are. Likewise, I don’t think that requiring command and sec time is going to guarantee that new nukies know how their mechanics work; you can have command shifts which don’t teach you anything relevant to being a nukie commander and sec shifts where you never shoot a gun. I think just having a flat 30 hour requirement would work the best.

**Comment by deltanedas** [comment]
whereas salv is surviving in space, killing dragons, healing yourself in battle, etc

**Comment by GoodWheatley** [comment]
> whereas salv is surviving in space, killing dragons, healing yourself in battle, etc

Or it’s just flying back and forth and looting magnet debris for the shift. And with how competitive getting the salvage tech job roundstart is already, requiring salvage playtime just means that only salvage mains will get to play nukies for potentially a month.

**Comment by deltanedas** [comment]
that means more sosa ops which is good

**Comment by GoodWheatley** [comment]
> that means more sosa ops which is good

How is letting a single player have a vastly higher chance to roll antags good for the game

**Comment by LankLTE** [comment]
> Commander requiring operative time would suck because of how low the chances of rolling nukie are. Likewise, I don’t think that requiring command and sec time is going to guarantee that new nukies know how their mechanics work; you can have command shifts which don’t teach you anything relevant to being a nukie commander and sec shifts where you never shoot a gun. I think just having a flat 30 hour requirement would work the best.

It requiring operative time may suck, but it's a good thing. I think like, literally half an hour would be fine, just to ensure that you _have_ played nukie at some point in the past and don't have people playing their first nukie game on commander. 
Everything else I agree with

**Comment by deltanedas** [comment]
yeah im not saying 10 hours nukie requirement for commander, that would be simply impossible

**Comment by Emisse** [comment]
commander should be like 1 hour since thats 1-2 nukie missions and thats enough time to be acquainted with it

**Comment by crazybrain23** [comment]
> commander should be like 1 hour since thats 1-2 nukie missions and thats enough time to be acquainted with it

If operative roletimers were based on time, there would be an incentive to waste time as nukies so you can unlock commander. Would be better to track number of times as operative?

**Comment by deltanedas** [comment]
then youd be incentivised to group suicide to get a new round quicker, which is much worse

**Comment by JoeHammad1844** [comment]
how about some hours as security aswell?
1 hour as salv and 1 hour as sec sounds reasonable to me

**Comment by metalgearsloth** [comment]
> For the nukie requirements, I think it'd be good if commander, maybe even agent, had a requirement for having played as nukie before, but it'd need to be really short to make sure it's actually reasonable to unlock. Like emisse said, 1-2 rounds is probably fine most of the time, it'd probably be enough to ensure that the commander can give some guidance to the team. 

Just make it like 10-15 minutes so there's no dramas about getting it after a round.

**Comment by deltanedas** [comment]
merg

**Comment by JoeHammad1844** [comment]
add some hos playtime to commander and CMO gameplay to agent

File: Resources/Prototypes/Roles/Antags/nukeops.yml [hunk 1]
```diff
@@ -4,17 +4,38 @@
...
   objective: roles-antag-nuclear-operative-objective
+  requirements:
+  - !type:OverallPlaytimeRequirement
+    time: 90000 # 25h
+  - !type:DepartmentTimeRequirement
+    department: Security
+    time: 18000 # 5h
+  - !type:DepartmentTimeRequirement
+    department: Command
+    time: 18000 # 5h
```
[the remark this answers is not in the document]
Comment (crazybrain23) [reply Resources/Prototypes/Roles/Antags/nukeops.yml:15]: Second point is that you gave the regular nukies a command playtime requirement but the commander doesn't have the command playtime requirement.

**Comment by metalgearsloth** [comment]
@VigersRay if you're going to work on your PRs ping us on discord.

**Comment by Geekyhobo** [comment]
@VigersRay Is it alright If I try and revive this PR? Wanna make this happen

**Comment by Geekyhobo** [comment]
and #17928

**Comment by Morb0** [comment]
> @VigersRay Is it alright If I try and revive this PR? Wanna make this happen

You can, just add him as an author

**Comment by deltanedas** [comment]
revive it address reviews and in cl do
`:cl: VigersRay, Geekyhobo`

**Comment by crazybrain23** [comment]
> @VigersRay Is it alright If I try and revive this PR? Wanna make this happen

https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/checking-out-pull-requests-locally

If you run `git fetch upstream pull/17929/head:nukeops_role_restrictions` you'll have a local copy of this PR that you can make changes to. It keeps their commit history and makes a co-authored PR.

**Comment by deltanedas** [comment]
bruh real every time i want to compile or ~~steal~~ revive someone pr i add their fork as a remote :trollface:

**Comment by deltanedas** [comment]
ok @Geekyhobo you took too long this is my pr now

**Comment by Geekyhobo** [comment]
@deltanedas Go for it, I dont really have the time due to school.

**Pull request closed without merging**